Source/NSSpeechRecognizer.m

```diff
@@ -23,8 +23,279 @@
 */
 
 #import <AppKit/NSSpeechRecognizer.h>
+#import <AppKit/NSApplication.h>
+#import <Foundation/NSDistantObject.h>
+#import <Foundation/NSString.h>
+#import <Foundation/NSDictionary.h>
+#import <Foundation/NSArray.h>
+#import <Foundation/NSThread.h>
+#import <Foundation/NSError.h>
+#import <Foundation/NSConnection.h>
+#import <Foundation/NSDistributedNotificationCenter.h>
+#import <Foundation/NSDebug.h>
+#import <Foundation/NSUUID.h>
+#import "GSFastEnumeration.h"
+#import "AppKit/NSWorkspace.h"
 
-@implementation NSSpeechRecognizer
+id   _speechRecognitionServer = nil;
+BOOL _serverLaunchTested = NO;
+
+#define SPEECH_RECOGNITION_SERVER @"GSSpeechRecognitionServer"
+
+@interface NSObject (SpeechRecognitionServerPrivate)
+
+- (void) addToBlockingRecognizers: (NSString *)s;
+- (void) removeFromBlockingRecognizers: (NSString *)s;
+- (BOOL) isBlocking: (NSString *)s;
+- (void) addClient;
 
 @end
 
+@implementation NSSpeechRecognizer
+
++ (void) initialize
+{
+  if (self == [NSSpeechRecognizer class])
+    {
+      // Test for an existant server...
+      _speechRecognitionServer =
+        [NSConnection rootProxyForConnectionWithRegisteredName: SPEECH_RECOGNITION_SERVER
+                                                          host: nil];
+
+      // if none exists, start one.  We will connect with it in init.
+      if (nil == _speechRecognitionServer)
+        {
+          NSWorkspace *ws = [NSWorkspace sharedWorkspace];
+          [ws launchApplication: SPEECH_RECOGNITION_SERVER
+                       showIcon: NO
+                     autolaunch: NO];
+        }
+    }
+}
+
+- (void) _restartServer
```

Comment:
This only reconnects but does not restart the server.

Reply:
It will now restart the server if it is not started.